script.js

```diff
@@ -1 +1,341 @@
+};
+
+// creating display Message that disappear after 3 seconds
+const messageDisplay = (boardElement) => {
+  const messageDisplayDiv = document.createElement('div');
+  messageDisplayDiv.classList.add('message-display-div, display');
+  messageDisplayDiv.innertext = 'Its a match!';
+  boardElement.appendChild(messageDisplayDiv);
+
+  // prevent user from having the ability to click
+  abilityToClick = false;
+  // using setTimeOut() to make message disappear after 3 seconds
+  setTimeout(() => {
+    boardElement.removeChild(messageDisplayDiv);
+    abilityToClick = true;
+    // change game message
+    gameMessage('Please wait 3 Seconds...<br> Please press another card to match!');
+  }, 3000);
+};
+// No match by user
+const noMatch = (firstSquare, secondSquare) => {
+  abilityToClick = false;
+  setTimeout(() => {
+    firstSquare.innerText = '';
+    secondSquare.innerText = '';
+    abilityToClick = true;
+  }, 3000);
+};
+
+// Timer display
+let timerID = 0;
+const countDownEl = document.createElement('text');
+countDownEl.setAttribute('class', 'display');
+let currentTime = 0;
+// countDown function
+const countDown = () => {
+  currentTime -= 1;
+  countDownEl.innerHTML = `Time Remaining: ${currentTime}`;
+  console.log(currentTime);
+  if (currentTime === 0) {
+    clearInterval(timerID);
+    // eslint-disable-next-line no-alert
+    alert(`Game Over! Your total matched is ${totalMatched}`);
+    // eslint-disable-next-line no-restricted-globals
+    location.reload();
+  }
+};
+// change matched display
+const changeMatchedDisplay = (firstSquare, secondSquare) => {
+  firstSquare.innerHTML = '✅';
+  secondSquare.innerHTML = '✅';
+};
+// Score display
+const scoreMessageDiv = document.createElement('div');
+scoreMessageDiv.setAttribute('class', 'display');
+
+// create input box for user to input name
+const nameInput = () => {
+  // overall div for displaying of name
+  const userDisplayDiv = document.createElement('div');
+  userDisplayDiv.setAttribute('class', 'display');
+  userDisplayDiv.setAttribute('id', 'display');
+  // textbox for input
+  const inputBox = document.createElement('input');
+  // giving the box, attribute and adding placeholder
+  inputBox.setAttribute('type', 'text');
+  inputBox.setAttribute('placeholder', 'Please Input Name');
+  inputBox.setAttribute('class', 'display');
+  // setting id so we can retrieve the value later
+  inputBox.setAttribute('id', 'inputUser');
+  // creating the submit button
+  const submitButton = document.createElement('button');
+  submitButton.innerHTML = 'Submit';
+  // adding an eventlistner to the submitbutton to retrieve userInput
+  submitButton.addEventListener('click', () => {
+    const userInput = document.querySelector('#inputUser');
+    gameInstructionDiv.innerHTML = `Hey there ${userInput.value}! Please click on a card to begin`;
+    inputBox.value = '';
+    gameMode = 'play';
+    // eslint-disable-next-line no-use-before-define
+    gameInit();
```

Comment:
Seems like an anti-pattern to call `gameInit` multiple times. `gameInit` is meant to be the function that sets up the DOM and game variables, not the function that controls the gameplay. In general, try to separate the functions that initialise and manipulate DOM elements from functions that define the logic of the game. This will make your code easier to read and more manageable.